### ISSUE TITLE
Potential risk when batch_size is greater than 1

### ISSUE THREAD

**Issue by wk910930** [torrvision/crfasrnn#107]
Dear @bittnt ,

I met strange issues when setting the batch_size greater than 3. Sometimes it complains `illegal memory access was encountered`, while sometimes the training loss becomes `nan` immediately.

I was wondering if you have implemented a new multi_stage_meanfield_layer that supports arbitrary batch_size?

Thanks!
Kun

**Comment by sadeepj** [comment]
Hi Kun,

Our current implementation only supports batch_size 1. In fact, there's a log message in https://github.com/torrvision/caffe/blob/70856cd28a10e2592b4d5ba1ae05cf6b59824a37/src/caffe/layers/multi_stage_meanfield.cpp#L49 mentioning this. We will make this an error/warning to make it clearer. Thanks for reporting the issue.
